website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md

```diff
@@ -123,3 +133,20 @@ If your pipeline uses non-Python packages (e.g. packages that require installati
         --setup_file /path/to/setup.py
 
 **Note:** Because custom commands execute after the dependencies for your workflow are installed (by `pip`), you should omit the PyPI package dependency from the pipeline's `requirements.txt` file and from the `install_requires` parameter in the `setuptools.setup()` call of your `setup.py` file.
+
+## Pre-building SDK container image
+
+In the pre-building step, we install pipeline dependencies on the container image prior to the job submission. This would speed up the pipeline execution.\
+To use pre-building the dependencies from `requirements.txt` on the container image. Follow the steps below.
+1. Provide the container engine. We support `local_docker` and `cloud_build`(requires a GCP project with Cloud Build API enabled).
+
+       --prebuild_sdk_container_engine <execution_environment>
+2. To pass a base image for pre-building dependencies, enable this flag. If not, apache beam's base image would be used.
+
+       --sdk_container_image <location_to_base_image>
+3. To push the container image, pre-built locally with `local_docker` , to a remote repository(eg: docker registry), provide URL to the remote registry by passing
+
+       --docker_registry_push_url <IMAGE_URL>
```

Comment:
I am confused - what is a sample value of this param? Is it supposed to be the image name+tag or just the registry?

Reply:
It should be docker registry. We generate the image tag. Image name is coded as `beam_python_prebuilt_sdk`[1].

May be it can worded as `--docker_registry_push_url` <registry_URL>

[1] https://github.com/apache/beam/blob/2a23166f738182f4553a79b8fc44c46e10705e76/sdks/python/apache_beam/runners/portability/sdk_container_builder.py#L85

Reply:
Can we give an example of the expected value? As a user reading this doc it is still not obvious.

Reply:
We can add an example. Also let me see if I can make the wording more simpler